src/libs/HttpUtils.js

```diff
@@ -18,6 +19,15 @@ Onyx.connect({
     callback: val => shouldFailAllRequests = (val && _.isBoolean(val.shouldFailAllRequests)) ? val.shouldFailAllRequests : false,
 });
 
+// getEnvironment promise returns the environment based on ENVIRONMENT Config for web/desktop.
+// Additionally, for iOS/android betaChecker.isBetaBuild() does the version check
+// environment is initialized as null as the default set in CONFIG.js always takes precedence
+let environment = null;
+Environment.getEnvironment()
+    .then((env) => {
```

Comment:
Do you suggest to then await like below. 

`const environment = await Environment.getEnvironment();`

The only issue with await was it shall make the xhr execution wait in HttpUtils till  promises returns a value. This may slow down the app starting in android environment since beta check is a network call. Earlier with null we will may see a null value in staging during environment initialization period.

Reply:
I think we are not using `await` in our codebase.

When is this being called @smrutiparida? This is called just once and before being resolved it is null, right? In that case we will use the default which is production?

What if this is in dev, that is not influenced and we will use whatever is in the .env right?

Reply:
@mountiny 
> When is this being called @smrutiparida? This is called just once and before being resolved it is null, right? In that case we will use the default which is production?

Yes, we will fall back to production API_URL

> What if this is in dev, that is not influenced and we will use whatever is in the .env right?

Yes. 
One note here is that presently .env do not have STAGING_EXPENSIFY_URL and STAGING_SECURE_EXPENSIFY_URL configuration. The present values are hard-coded in Config.js. We need to init the values in .env in order to use them in dev environment.

Reply:
> Yes, we will fall back to production API_URL

This will have to reset everytime the app is killed and opened again, right? If I just put it in a background and come back the value will be still "cached"

Reply:
I was thinking that we create a Middleware that switches the URL based on the ENV. But that seems not possible. Our Middleware applies after the request is sent to the server. 

@smrutiparida Could you please share some suggestions on what are the options? 

> Yes, we will fall back to production API_URL

How? the local `environment` variable initialized with `null` value. Until the promise is resolved it will be null. 

Thus, if `shouldUseStagingServer` is set, requests will go to staging even if a user is using the PROD app. This sounds wrong. Right?

Reply:
>How? the local environment variable is initialized with null value. Until the promise is resolved it will be null.
> Thus, if shouldUseStagingServer is set, requests will go to staging even if a user is using the PROD app. This sounds wrong. Right?

No, with below logic user can only be in staging with envrionment set to staging.
if (environment === CONST.ENVIRONMENT.STAGING && shouldUseStagingServer) {.... }

Reply:
Oh Correct. I don't know why I got confused.

Reply:
`shouldUseStagingServer` should never be set to truthy value in prod app, the toggle does not exist in there I believe.